app/assets/stylesheets/application.css

```diff
@@ -659,7 +659,8 @@ li .comment_parent_tree_line.score_shown {
 li .comment_parent_tree_line.can_flag {
 	top: 56px;
 }
-li .comment_parent_tree_line.no_children {
+/* hide for comments that don't have children */
+li:not(:has(> .comments > li)) > .comment > .comment_parent_tree_line {
```

Comment:
That's really nice, `:has` wasn't available when this was first written, but it's great that you saw the opportunity to simplify the amount of state the template has to juggle.